olivertwist/config/parser.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+"""Document __init__.py here.
+
+Copyright (C) 2020, Auto Trader UK
+Created 22. Dec 2020 19:36
+
+"""
+import os.path
+import yaml
+
+from dataclasses import dataclass
+from typing import List, Optional
+
+try:
+    from yaml import CSafeLoader as YamlLoader
+except ImportError:
+    from yaml import SafeLoader as YamlLoader
+
+from dataclasses_jsonschema import JsonSchemaMixin, ValidationError
+
+
+@dataclass
+class RuleConfig(JsonSchemaMixin):
+    id: str
+    enabled: bool
+
+
+@dataclass
+class Config(JsonSchemaMixin):
+    universal: List[RuleConfig]
+
+    def get_disabled_rule_ids(self) -> List[str]:
+        return [r.id for r in self.universal if r.enabled is False]
+
+
+class InvalidConfigException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+class ConfigParser:
+    @classmethod
+    def parse(cls, config_file_path) -> Config:
+        try:
+            with open(config_file_path, "rb") as handle:
+                yaml_config_dict = yaml.load(
+                    handle.read().decode("utf-8"), Loader=YamlLoader
+                )
+                return Config.from_dict(yaml_config_dict)
+        except ValidationError as e:
+            raise InvalidConfigException(e)
+
+    @classmethod
+    def from_default_path(cls) -> Optional[Config]:
+        default_path = "./oliver_twist.yml"
+        if os.path.isfile(default_path):
+            return cls.parse(default_path)
+        else:
+            return None
```

Comment:
Feels as though when calling this func, you expect a config to be returned. If it can't be returned, then this should throw and exception, which is caught in `main`.

Returning `None` is tied to the specific way it is currently being used, which may trip us up in the future. I'd move that logic to `main`.